static/js/reducers/dashboard.js

```diff
@@ -13,39 +14,48 @@ import {
   FETCH_PROCESSING,
   FETCH_SUCCESS,
 } from '../actions';
-import type { DashboardState } from '../flow/dashboardTypes';
+import type {
+  DashboardsState,
+  DashboardState,
+} from '../flow/dashboardTypes';
 import type { Action } from '../flow/reduxTypes';
 
-const INITIAL_DASHBOARD_STATE: DashboardState = {
+export const INITIAL_DASHBOARD_STATE: DashboardState = {
   programs: []
 };
 
-export const dashboard = (state: DashboardState = INITIAL_DASHBOARD_STATE, action: Action) => {
+const INITIAL_DASHBOARDS_STATE: DashboardsState = {};
+
+const updateDashboardState = (state, username, update) => (
+  _.merge({}, state, {[username]: update })
```

Comment:
Basically I implemented it this way because there are situations where the old state looks something like this:

```js
[username]: {
  programs: [],
}
```

and we want to, for instance, update the state to have a `fetchProcessing` key on it, but we want to keep around the `programs` array. I hadn't considered that this might lead to some strange stuff with the dashboard being merged together.

In particular if you do something like (just tested this out in the node repl):

```js
let obj = { username: { programs: [ {foo: 'bar' }], otherProp: 'potato' } };
_.merge({}, obj, {username: { programs: [] } }) 
// { username: { programs:[ {foo:"bar"} ], otherProp:"potato" } }
```

the old 'programs' value isn't overwritten by the new empty array.

I currently use `R.dissoc` to delete any user specific state when we handle the `CLEAR_DASHBOARD` action, I could do the same thing for some of the other actions too.